cldfbench_segbo.py

```diff
@@ -2,44 +2,112 @@
+        table = ds.add_component('ParameterTable')
+
+        # languages.csv
+        ds.add_component('LanguageTable', *self.languageTableProperties)
+
+        # inventories.csv
         table = ds.add_table(
-            'segbo_with_glottolog.csv',
+            'inventories.csv',
             {'name': 'ID', 'propertyUrl': "http://cldf.clld.org/v1.0/terms.rdf#id"},
-            {'name': 'InventoryID'},
+            {'name': 'Language_ID', 'propertyUrl': "http://cldf.clld.org/v1.0/terms.rdf#languageReference"},
+            {'name': 'Language_Name', 'propertyUrl': "http://cldf.clld.org/v1.0/terms.rdf#name"},
+            {'name': 'Source', 'propertyUrl': 'http://cldf.clld.org/v1.0/terms.rdf#source', 'separator': ';'},
         )
         table.tableSchema.primaryKey = ['ID']
+        ds.add_foreign_key('ValueTable', 'Inventory_ID', 'inventories.csv', 'ID')
 
     def cmd_makecldf(self, args):
-        """
-        Convert the raw data to a CLDF dataset.
-
-        >>> args.writer.objects['LanguageTable'].append(...)
-        """
         self.create_schema(args.writer.cldf)
 
+        # values.csv
         counter = 1
         for row in self.raw_dir.read_csv(
-            self.raw_dir / 'segbo' / 'data' / 'segbo_with_glottolog.csv',
+            self.raw_dir / 'segbo' / 'data' / 'SegBo database - Phonemes.csv',
             dicts=True,
         ):
-            value = {
-                **{"ID": str(counter)},
-                **dict(row)
-            }
-            args.writer.objects['segbo_with_glottolog.csv'].append(value)
+            args.writer.objects['ValueTable'].append({
+                'ID': str(counter),
+                'Parameter_ID': str(counter),
+                'Inventory_ID': row['InventoryID'],
+                'Language_ID': row['BorrowingLanguageGlottocode'],
+                'Source_Language_ID': row['SourceLanguageGlottocode'],
+                'Inventory_ID': row['InventoryID'],
+                'Value': row['BorrowedSound'],
+                'Source': '', # TODO 
+                **{ k: row[k] for k in self.valueTableProperties}
+            })
+            # parameters.csv
+            args.writer.objects['ParameterTable'].append({
+                'ID': str(counter),
+                'Name': row['BorrowedSound'],
+                'Description': '', # TODO
+                # TODO more features, do not exist in raw data
+            })
             counter += 1
+        
+        # languages.csv
+        language_ids = list(map(lambda row: row['Language_ID'], args.writer.objects['ValueTable']))
+        for row in self.raw_dir.read_csv(
+            self.raw_dir / 'segbo' / 'data' / 'glottolog_languoid.csv' / 'languoid.csv',
+            dicts=True,
+        ):
+            if row['id'] in language_ids:
+                args.writer.objects['LanguageTable'].append({
+                    'ID': row['id'],
+                    'Name': row['name'],
+                    'Macroarea': '', # TODO, do not exist in raw data
```

Comment:
You could fetch these from Glottolog. `cldfbench` provides a `pyglottolog.Glottolog` instance as `args.glottolog.api`, which you could use to lookup the glottocodes.